src/Composer/ComposerizeDrupalCommand.php

```diff
@@ -255,9 +255,25 @@ protected function mergeTemplateGitignore()
     protected function requireDrupalCore($root_composer_json)
     {
         $version_constraint = DrupalInspector::getVersionConstraint($this->drupalCoreVersion, $this->input->getOption('exact-versions'));
-        $root_composer_json->require->{'drupal/core'} = $version_constraint;
-        $this->getIO()
-            ->write("<info>Added drupal/core $version_constraint to requirements.</info>");
+        if (version_compare($this->drupalCoreVersion, '8.8.0', '<')) {
+            $root_composer_json->require->{'drupal/core'} = $version_constraint;
+            $this->getIO()
+                ->write("<info>Added drupal/core $version_constraint to requirements.</info>");
+        } else {
+            $root_composer_json->require->{'drupal/core-recommended'} = $version_constraint;
+            $this->getIO()
+                ->write("<info>Added drupal/core-recommended $version_constraint to requirements.</info>");
+
+            unset($root_composer_json->require->{'drupal-composer/drupal-scaffold'});
+            $root_composer_json->require->{'drupal/core-composer-scaffold'} = $version_constraint;
+            $root_composer_json->extra->{"drupal-scaffold"}->{"locations"}{"web-root"} = "./$this->drupalRootRelative";
```

Comment:
Let's just add this to the template.

Reply:
Since drupal/core-composer-scaffold version should be same as Drupal core, I added only the drupal-scaffold section to the template.